### PR TITLE
Dispatcher enhancement

### PR DESCRIPTION
**Pull request by dongasai**
Hello!

Type: new feature

In raising this pull request, I confirm the following:

- [ ] I have read and understood the Contributing Guidelines
- [ ] I have checked that another pull request for this purpose does not exist
- [ ] I wrote some tests for this PR

Small description of change: Dispatcher enhancement, Allow interrupt action, Produce more of the possible.

Thanks

### CODE REVIEW

File: phalcon/dispatcher.zep [hunk 1]
```diff
@@ -199,6 +201,22 @@ abstract class Dispatcher implements DispatcherInterface, InjectionAwareInterfac
 	}
 
 	/**
+	 * Returns the canAction
+	 */
+	public function getCanAction() -> boolean
```
Comment (Jurigag) [inline phalcon/dispatcher.zep]: isCanAction

**Comment by virgofx** [comment]
I don't think this is necessary...

**Comment by sergeyklay** [comment]
I agree with @virgofx

**Comment by dongasai** [comment]
@virgofx  @sergeyklay This is a can enhance the Dispatcher in the MVC mode of Dispatch ,I use the sample is as follows:
`
<?php

class Test9Controller extends Phalcon\Mvc\Controller
{

    public  function initialize()
    {
        # For some reason, are not allowed to perform indexAction
        $this->dispatch->setCanAction(false);
    }

    public function indexAction()
    {
        # code 
    }

    /**
     * noCallAction
     */
    public function noCallAction()
    {
       # Perform other operations
    }

}
`

**Comment by dongasai** [comment]
It solves the: Verifies the permissions before performing the **_Action_** will affect the operation of the Action, but can't abandon **_afterDispatch_**. **_AfterExecuteRoute_** operation

**Comment by dongasai** [comment]
In the **_MVC_** structure, where did you perform the **_acl_** ? @virgofx  **Route**(_Before the new Controller_)?**dispatch:afterInitialize**( _Controller initialize_)?or  **The Action begins**(_This is not a right choice!_)
@sergeyklay

**Comment by sergeyklay** [comment]
@dongasai It seems to me you can address this by using ACL

**Comment by virgofx** [comment]
@dongasai Refer to phalcon documentation on dispatch forwarding. It will address exactly what you need. You can hook an ACL anytime or custom logic and forward to another action/controller as needed.

**Comment by dongasai** [comment]
@sergeyklay  First of all, this is not for enhanced authentication, this is the enhancement of dispatch,Then, if you ACL to return a false, you should know how to deal with the logic of it.
As @virgofx  said,Controller forwarding,But I don't think it's a good solution

**Comment by dongasai** [comment]
And forwarding will lead to continue the while

**Pull request closed without merging**